#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros_utils'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()

normalize_enum = gen.enum([gen.const("normalize_area", str_t, "normalize_area", "normalize_area"),
                           gen.const("normalize_height", str_t, "normalize_height", "normalize_height")],
                          "normalize")

plane_enum = gen.enum([gen.const("xy", str_t, "xy", "xy"),
                       gen.const("flipped_xy", str_t, "flipped_xy", "flipped_xy"),
                       gen.const("yz", str_t, "yz", "yz"),
                       gen.const("flipped_yz", str_t, "flipped_yz", "flipped_yz"),
                       gen.const("zx", str_t, "zx", "zx"),
                       gen.const("flipped_zx", str_t, "flipped_zx", "flipped_zx")],
                      "plane")
gen.add("normalize_method", str_t, 0, "", "normalize_area", edit_method=normalize_enum)
gen.add("normalize_value", double_t, 0, "", 1.0, 0.001, 10.0)
gen.add("cut_plane", str_t, 0, "", "xy", edit_method=plane_enum)
gen.add("sampling_num", int_t, 0, "", 100, 10, 1000)
exit (gen.generate (PACKAGE, "jsk_pcl_ros_utils", "PoseWithCovarianceStampedToGaussianPointCloud"))
